compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -530,10 +530,9 @@ static LogicalResult setWarpReductionConfig(func::FuncOp entryPoint,
 }
 
 /// Returns true if the index map represents a transpose that benefits from
-/// shared mem. Currently supports 2D transposes.
+/// shared mem.
 static bool isSharedMemTranspose(AffineMap indexMap) {
-  if (!indexMap.isEmpty() && indexMap.isPermutation() &&
-      indexMap.getNumInputs() == 2) {
+  if (!indexMap.isEmpty() && indexMap.isPermutation()) {
```

Comment:
This is probably a follow up but it would be interesting to also support some projected permutation as well

Reply:
What is a projected permutation?

Reply:
it a permutation where some dimension can be dropped from the results.

Reply:
Got it. Yeah I'll do this in a followup. Lets chat more about it after.